tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/create/CreateApplicationOnBuilderImageTest.java

```diff
@@ -102,6 +107,40 @@ public void testCreateApplicationBasedOnBuilderImage() {
 				+ " of services: " + services.size(), services.size() == 1);
 	}
 	
+    @Test
+    public void validateJBIDE22704() {
+        new NewOpenShift3ApplicationWizard().openWizardFromExplorer();
+        
+        BuilderImageApplicationWizardHandlingTest.nextToBuildConfigurationWizardPage();
+        
+        applicationName = new LabeledText("Name: ").getText();
+        
+        assertNotNull(applicationName);
+        assertTrue(applicationName.length() > 0);
+        
+        new WaitUntil(new WidgetIsEnabled(new NextButton()));
+
+        /*
+         * switch to the Deployment page
+         */
+        new NextButton().click();
+        new WaitUntil(new WidgetIsEnabled(new BackButton()));
+        
+        int numberofEnvironmentVariables = new DefaultTable().rowCount();
+        assertTrue(numberofEnvironmentVariables > 0);
+        
+        /*
+         * switch to the Routing page
+         */
+        new NextButton().click();
+        new WaitUntil(new WidgetIsEnabled(new BackButton()));
+        
+        int numberOfServicePorts = new DefaultTable().rowCount();
+        assertTrue(numberOfServicePorts > 0);
+        
+        new CancelButton().click();
```

Comment:
Use waiting to wait while shell is available and then while any job is running to be sure everything has finished as expected and there is no hanging shell. Tests are sometimes pretty fast and it could fail later if there would not be waiting and shell would be available a bit longer or any job would be running there.